service/storage_service.cc

```diff
@@ -682,6 +682,8 @@ void storage_service::bootstrap() {
         // Wait until we know tokens of existing node before announcing replacing status.
         set_mode(mode::JOINING, sprint("Wait until local node knows tokens of peer nodes"), true);
         _gossiper.wait_for_range_setup().get();
+
+      if (!is_repair_based_node_ops_enabled()) {
```

Comment:
no

Reply:
So we cannot mark the issue as fixed by this PR.

Reply:
Well, I doubt we will fix the path with repair based node ops disabled.

Reply:
We could change storage service listeners to wait for mark_alive pings before they change the token metadata.

Reply:
The listeners runs inside gossip loop. We should not block the loop waiting for the ping to successful.

Alternatively, we can use all the code path here but use streaming for Step 7 to sync data instead using repair to sync.

Reply:
The storage service listeners could move to background (serializing) and don't block the gossiper.

Reply:
> > > Looks good. But every time we enable repair based operations we revert the, later. What testing was done to make sure this doesn't happen this time?
> > 
> > 
> > This patch is more about the control plane. The data plane that uses repair to sync data does not change. I have tested with dtest. This patch does not solve any latency issues during replace.
> 
> Does it mean that the decision whether to use repair or not does not change (and it will still not be used by default?)



> > > Looks good. But every time we enable repair based operations we revert the, later. What testing was done to make sure this doesn't happen this time?
> > 
> > 
> > This patch is more about the control plane. The data plane that uses repair to sync data does not change. I have tested with dtest. This patch does not solve any latency issues during replace.
> 
> Does it mean that the decision whether to use repair or not does not change (and it will still not be used by default?)



> > > Looks good. But every time we enable repair based operations we revert the, later. What testing was done to make sure this doesn't happen this time?
> > 
> > 
> > This patch is more about the control plane. The data plane that uses repair to sync data does not change. I have tested with dtest. This patch does not solve any latency issues during replace.
> 
> Does it mean that the decision whether to use repair or not does not change (and it will still not be used by default?)

Yes, exactly.